src/openms/include/OpenMS/ANALYSIS/RNPXL/ModifiedPeptideGenerator.h

```diff
@@ -49,25 +53,54 @@ namespace OpenMS
     * @brief Modifications can be generated and applied to AASequences. 
     */
 
-  public
-:
+  public:
+    using MapToResidueType = boost::container::flat_map<const ResidueModification*, const Residue*>;
+
+      /**
+      * @brief Retrieve modifications from strings
+      * 
+      * @param modNames The list of modification names
+      * @return A map of modifications and associated residue
+      * ResidueModifications are referenced by Residues in AASequence objects. Every time an AASequence object
+      * with modifications is constructed, it needs to query if the (modified) Residue is already
+      * registered in ResidueDB. This implies a lock of the whole db. To make modified peptide generation lock-free, we
+      * query and cache all modified residues once so we can directly apply them without further queries.
+      */
+    static MapToResidueType getModifications(const StringList& modNames);
+
     // Applies fixed modifications to a single peptide
-    static void applyFixedModifications(const std::vector<const ResidueModification*>& fixed_mods, AASequence& peptide);
+    static void applyFixedModifications(
+      const MapToResidueType& fixed_mods, 
+      AASequence& peptide);
 
     // Applies variable modifications to a single peptide. If keep_original is set the original (e.g. unmodified version) is also returned
     static void applyVariableModifications(
-     const std::vector<const ResidueModification*>& var_mods, 
+     const MapToResidueType& var_mods, 
```

Comment:
this wont fly in pyOpenMS I think, either

- use the old interface for Python
- delete the interface
- create a small struct around the type:
```
struct MapToResidue {boost::container::flat_map<const ResidueModification*, const Residue*> rmap;}
```

Reply:
I found a way that does not need the map and should be compatible with pyOpenMS